src/server.c

```diff
@@ -129,23 +130,63 @@ void serverLogRaw(int level, const char *msg) {
     } else {
         int off;
         struct timeval tv;
-        int role_char;
         pid_t pid = getpid();
 
         gettimeofday(&tv,NULL);
         struct tm tm;
         nolocks_localtime(&tm,tv.tv_sec,server.timezone,server.daylight_active);
-        off = strftime(buf,sizeof(buf),"%d %b %Y %H:%M:%S.",&tm);
-        snprintf(buf+off,sizeof(buf)-off,"%03d",(int)tv.tv_usec/1000);
-        if (server.sentinel_mode) {
-            role_char = 'X'; /* Sentinel. */
-        } else if (pid != server.pid) {
-            role_char = 'C'; /* RDB / AOF writing child. */
-        } else {
-            role_char = (server.masterhost ? 'S':'M'); /* Slave or Master. */
+        switch(server.log_timestamp_format) {
+            case LOG_TIMESTAMP_DEFAULT:
+                off = strftime(buf,sizeof(buf),"%d %b %Y %H:%M:%S.",&tm);
+                snprintf(buf+off,sizeof(buf)-off,"%03d",(int)tv.tv_usec/1000);
+                break;
+
+            case LOG_TIMESTAMP_ISO8601:
+                off = strftime(buf,sizeof(buf),"%Y-%m-%dT%H:%M:%S.",&tm);
+                snprintf(buf+off,sizeof(buf)-off,"%03d",(int)tv.tv_usec/1000);
+                char tzbuf[6];
+                strftime(tzbuf, sizeof(tzbuf), "%z", &tm);
+                char tz_formatted[7];
+                snprintf(tz_formatted, sizeof(tz_formatted), "%c%c%c:%c%c", tzbuf[0], tzbuf[1], tzbuf[2], tzbuf[3], tzbuf[4]);
+                strncat(buf, tz_formatted, sizeof(buf) - strlen(buf) - 1);
```

Comment:
Good, but maybe we can do it with only one `snprintf` for milliseconds and time zone?

```suggestion
                char tzbuf[6];
                strftime(tzbuf, sizeof(tzbuf), "%z", &tm);
                snprintf(buf + off, sizeof(buf) - off, "%03d%c%c%c:%c%c",
                         (int)tv.tv_usec/1000,
                         tzbuf[0], tzbuf[1], tzbuf[2], tzbuf[3], tzbuf[4]);
```

Reply:
Thanks. Updated as your suggestion.